fvm/transactionInvoker.go

```diff
@@ -75,7 +75,15 @@ func (i *TransactionInvoker) Process(
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
 		}
-		if mergeError := parentState.MergeState(childState, sth.EnforceInteractionLimits()); mergeError != nil {
+
+		// backwards compatibility to enable rolling deploy
+		enforceLimits := sth.EnforceInteractionLimits()
+		if (ctx.Chain.ChainID() == flow.Mainnet && blockHeight > uint64(36419249)) ||
+			(ctx.Chain.ChainID() == flow.Testnet && blockHeight > uint64(78354421)) {
```

Comment:
I wonder if we need these block heights?

Reply:
I would say yes if someone wants to rerun old blocks.